lib/evmone_precompiles/ripemd160.cpp

```diff
@@ -0,0 +1,219 @@
+    // f₁(x, y, z) = x ⊕ y ⊕ z
+    [](uint32_t x, uint32_t y, uint32_t z) noexcept { return x ^ y ^ z; },
+
+    // f₂(x, y, z) = (x ∧ y) ∨ (¬x ∧ z) ⇔ ((y ⊕ z) ∧ x) ⊕ z
+    [](uint32_t x, uint32_t y, uint32_t z) noexcept { return ((y ^ z) & x) ^ z; },
+
+    // f₃(x, y, z) = (x ∨ ¬y) ⊕ z
+    [](uint32_t x, uint32_t y, uint32_t z) noexcept { return (x | ~y) ^ z; },
+
+    // f₄(x, y, z) = (x ∧ z) ∨ (y ∧ ¬z) ⇔ ((x ⊕ y) ∧ z) ⊕ y
+    [](uint32_t x, uint32_t y, uint32_t z) noexcept { return ((x ^ y) & z) ^ y; },
+
+    // f₅(x, y, z) = x ⊕ (y ∨ ¬z)
+    [](uint32_t x, uint32_t y, uint32_t z) noexcept { return x ^ (y | ~z); },
+};
+
+/// Added constants.
+constexpr uint32_t constants[L][R] = {
+    {
+        0,
+        0x5a827999,
+        0x6ed9eba1,
+        0x8f1bbcdc,
+        0xa953fd4e,
+    },
+    {
+        0x50a28be6,
+        0x5c4dd124,
+        0x6d703ef3,
+        0x7a6d76e9,
+        0,
+    },
+};
+
+
+/// Selection of message word.
+constexpr size_t word_index[L][N] = {
+    {
+        /*r ( 0..15) = */ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15,  //
+        /*r (16..31) = */ 7, 4, 13, 1, 10, 6, 15, 3, 12, 0, 9, 5, 2, 14, 11, 8,  //
+        /*r (32..47) = */ 3, 10, 14, 4, 9, 15, 8, 1, 2, 7, 0, 6, 13, 11, 5, 12,  //
+        /*r (48..63) = */ 1, 9, 11, 10, 0, 8, 12, 4, 13, 3, 7, 15, 14, 5, 6, 2,  //
+        /*r (64..79) = */ 4, 0, 5, 9, 7, 12, 2, 10, 14, 1, 3, 8, 11, 6, 15, 13,  //
+    },
+    {
+        /*r′( 0..15) = */ 5, 14, 7, 0, 9, 2, 11, 4, 13, 6, 15, 8, 1, 10, 3, 12,  //
+        /*r′(16..31) = */ 6, 11, 3, 7, 0, 13, 5, 10, 14, 15, 8, 12, 4, 9, 1, 2,  //
+        /*r′(32..47) = */ 15, 5, 1, 3, 7, 14, 6, 9, 11, 8, 12, 2, 10, 0, 4, 13,  //
+        /*r′(48..63) = */ 8, 6, 4, 1, 3, 11, 15, 0, 5, 12, 2, 13, 9, 7, 10, 14,  //
+        /*r′(64..79) = */ 12, 15, 10, 4, 1, 5, 8, 7, 6, 2, 13, 14, 0, 3, 9, 11,  //
+    },
+};
+
+/// Amount for rotate left.
+constexpr int rotate_amount[L][N] = {
+    {
+        /* s ( 0..15) = */ 11, 14, 15, 12, 5, 8, 7, 9, 11, 13, 14, 15, 6, 7, 9, 8,  //
+        /* s (16..31) = */ 7, 6, 8, 13, 11, 9, 7, 15, 7, 12, 15, 9, 11, 7, 13, 12,  //
+        /* s (32..47) = */ 11, 13, 6, 7, 14, 9, 13, 15, 14, 8, 13, 6, 5, 12, 7, 5,  //
+        /* s (48..63) = */ 11, 12, 14, 15, 14, 15, 9, 8, 9, 14, 5, 6, 8, 6, 5, 12,  //
+        /* s (64..79) = */ 9, 15, 5, 11, 6, 8, 13, 12, 5, 12, 13, 14, 11, 8, 5, 6,  //
+    },
+    {
+        /* s′( 0..15) = */ 8, 9, 9, 11, 13, 15, 15, 5, 7, 7, 8, 11, 14, 14, 12, 6,  //
+        /* s′(16..31) = */ 9, 13, 15, 7, 12, 8, 9, 11, 7, 7, 12, 7, 6, 15, 13, 11,  //
+        /* s′(32..47) = */ 9, 7, 15, 11, 8, 6, 6, 14, 12, 13, 5, 14, 13, 13, 7, 5,  //
+        /* s′(48..63) = */ 15, 5, 8, 11, 14, 14, 6, 14, 6, 9, 12, 9, 12, 5, 15, 8,  //
+        /* s′(64..79) = */ 8, 5, 12, 9, 12, 5, 14, 6, 8, 13, 6, 5, 15, 13, 11, 11,  //
+    },
+};
+
+/// Converts native representation to/from little-endian.
+inline auto to_le(std::integral auto x) noexcept
```

Comment:
similar methods exist in `intx::le` namespace in case you are able to use intx.hpp here, see: https://github.com/erigontech/silkworm/blob/master/silkworm/core/common/endian.hpp

Reply:
On one hand I plan to improve intx's API for this. On the other hand, I wanted to stick to C++ std only for this file.